cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/adapter/AdapterService.java

```diff
@@ -106,6 +120,27 @@ protected void shutDown() throws Exception {
   }
 
   /**
+   * Deploy the given application template in the given namespace.
+   *
+   * @param namespace the namespace to deploy in
+   * @param templateName the name of the template to deploy
+   * @throws NotFoundException if the template was not found
+   * @throws IllegalArgumentException if the template is invalid
+   * @throws IOException if there was an error reading the template jar
+   * @throws TimeoutException if there was a timeout examining the template jar
+   */
+  public void deployTemplate(Id.Namespace namespace, String templateName)
+    throws NotFoundException, InterruptedException, ExecutionException, TimeoutException, IOException {
+    ApplicationTemplateInfo templateInfo = appTemplateInfos.get(templateName);
+    if (templateInfo == null) {
+      throw new NotFoundException("ApplicationTemplate", templateName);
+    }
+    // make sure we're up to date on template info
+    registerTemplates();
+    deployTemplate(namespace, templateInfo);
```

Comment:
Will this fail if a program of the previous version of the template-app is currently running (for ex, worker)? Should the user take care of stopping all adapters of the specific template before updating the template?